opensquirrel/merger/general_merger.py

```diff
@@ -169,10 +172,11 @@ def merge_single_qubit_gates(circuit: Circuit) -> None:  # noqa: C901
             continue
 
         # Accumulate consecutive Bloch sphere rotations
-        if isinstance(statement, BlochSphereRotation):
-            already_accumulated = accumulators_per_qubit[statement.qubit]
-            composed = compose_bloch_sphere_rotations(statement, already_accumulated)
-            accumulators_per_qubit[statement.qubit] = composed
+        instruction: Instruction = cast(Instruction, statement)
```

Comment:
Is the use of `cast` purely here for the purpose of code clarity? (which is fine by the way)

It is mainly used to deal with multiple possible return types or third-party software that isn't typed properly so that mypy doesn't complain on our side (at a minor performance cost). But here it is not necessary right?

Reply:
I think `mypy --strict` complains if you don't use it, because not all statements are instructions. You can also have comments. So here we're just saying that we know `statement` is of type Instruction.